js/twister_formatpost.js

```diff
@@ -379,196 +404,228 @@ function htmlFormatMsg(msg, mentions) {
                 if (p[i].a > -1) {
                     if (p[i].t === -1 || (p[i].t === 0 && p[i].a > i)) {
                         if (p[i].k > 1)
-                            html.push('<' + tag + '>' + Array(p[i].k).join(chr));
+                            msg.htmlEntities.push('<' + tag + '>' + Array(p[i].k).join(chr));
                         else
-                            html.push('<' + tag + '>');
+                            msg.htmlEntities.push('<' + tag + '>');
                     } else if (p[i].t === 1 || (p[i].t === 0 && p[i].a < i)) {
                         if (p[i].k > 1)
-                            html.push(Array(p[i].k).join(chr) + '</' + tag + '>');
+                            msg.htmlEntities.push(Array(p[i].k).join(chr) + '</' + tag + '>');
                         else
-                            html.push('</' + tag + '>');
+                            msg.htmlEntities.push('</' + tag + '>');
                     }
-                        strEncoded = '>' + (html.length - 1).toString() + '<';
-                        str = str.slice(0, p[i].i) + strEncoded + str.slice(p[i].i + p[i].k);
-                        l = strEncoded.length - p[i].k;
+                        htmlEntityEncoded = '>' + (msg.htmlEntities.length - 1).toString() + '<';
+                        msg.str = msg.str.slice(0, p[i].i) + htmlEntityEncoded + msg.str.slice(p[i].i + p[i].k);
+                        l = htmlEntityEncoded.length - p[i].k;
                         for (j = i + 1; j < p.length; j++)
                             p[j].i += l;
                 }
             }
         }
 
-        return str;
+        return msg;
     }
 
-    function htmlSplitCounter(str) {
-        html.push('<span class="splited-post-counter">' + str + '</span>');
+    function newHtmlEntityLink(template, urlTarget, urlName) {
+        template.href = urlTarget;
+        template.innerHTML = urlName;  // .innerHTML instead .text to allow markup inside [] of [url name](target)
```

Comment:
using innerHTML in newHtmlEntityLink is also possibly dangerous. are we sure it's always sanitized?
idea: how about two versions, like newHtmlEntityLinkFormatted and newHtmlEntityPlaintext? the second would use .text and we try using it unless absolutely necessary.